datastream-utils/src/main/java/com/linkedin/datastream/common/LogUtils.java

```diff
@@ -92,4 +95,44 @@ public static String logSummarizedTopicPartitionsMapping(List<String> partitions
             .append(logNumberArrayInRange(topicPartitionsMap.get(topicName))))
         .collect(Collectors.joining(", "));
   }
+
+  /**
+   * prints one log line for each string smaller than size limit
+   * @param  message
+   */
+  public static void logStringsUnderSizeLimit(Logger log, String message, String contextPrefix, int part, double sizeLimit) {
+    int bufferAdjustedSizeLimit = (int) (sizeLimit - BUFFER_1KB);
+    if (isLessThanSizeLimit(message, sizeLimit)) {
+      if (part == 1) {
+        log.info("{}={}", contextPrefix,  message);
+      } else {
+        log.info("{} (part {})={}", contextPrefix, part,  message);
```

Comment:
what happens if the contextPrefix is more than the 1KB Buffer, the log will be truncated. So in the public facing method, lets also check for the size of that contextPrefix string and fail if it exceeds 1KB Buffer.